docs/rfcs/017-asset-modified-cleanup.md

```diff
@@ -0,0 +1,140 @@
+  - Ingestion completion is needed so that we can check when the asset was last ingested, and whether the attached policy has changed in that time
+  - If there's an error, no need to do anything
+- Telling the difference between `Before` and `After` should be done with using the `Before` from the message, but the `After` be pulled from the database, to avoid issues with multiple reingests happening
+  - This would help to avoid issues where the asset is changed, then changed back afterwards, but due to `delay_seconds`, the `After` on the message clears up the correct derivatives
+
+### Logic
+
+#### Thumbs recalculation
+
+Within this, the most complex part of this is recalculating thumbnails.  The general approach to this will be to use the policy of the current asset, along with the thumbnails that currently exist within S3.  This will require a `ListBucket` operation from S3 which has a cost implication.  This could mean that checking 53 million images, would incur a cost of approximately $265
+
+Due to the cost, calls to `ListBucket` should be limited
+
+Thumbs should use the iiif-net library to check expected sizes of thumbnails for an image, and there needs to be some logic to not remove thumbs that are within 2-3 pixels of the expected to avoid off-by-one errors in thumbnail generation.  These off-by-one errors occur due to rounding errors in cantaloupe thumbs generation, more detgails can be found [here](https://github.com/dlcs/protagonist/pull/819)
+
+System thumbs will need to be left alone.  In order to make it so this doesn't differ from the system thumbs in engine, a  value needs to be created that can be used by both engine and the cleanup handler.  This is used by these projects through the options pattern in .Net. One of the ways to do this, is a parameter store variable that is exposed to the engine and orchestrator via an environment variable
+
+Finally, maxUnauth/Roles will need to be taken into account as storage in the bucket will have a prefix of `/open` or `/auth` based on this.
+
+#### Named query derivatives
+
+There are many different permutations of objects in named queries, so for now this is out of scope for cleanup.
+
+#### Update types
+
+There are a number of ways that an update to an asset can cause changes to stored derivatives that aren't tracked.  It can roughly be divided into 4 categories of change:
+
+- Delivery channel changed
+- Policy id changed
+- Roles changed
+- Policy data updated
+- Origin changes
+
+#### Delivery channels changed
+
+This becomes an issue when a delivery channel is removed from an asset, with the following implications:
+
+- iiif-img removed
+  - Stored iiif-img derivative needs to be removed
+  - `info.json` removed
+  - Asset can exist at both filename and `/original` path - though the `original` needs to be left if the file channel exists on the asset
+  - if `thumbs` not there, then `s.json` and thumbnails need removed as well 
```

Comment:
and remove `AssetApplicationMetadata`. I think this is implied by removal of `s.json` but it's explicit elsewhere